File: 01-10/Merge Sorted Array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        sorted_list = list()
        p1 = 0
        p2 = 0
        while p1 < m or p2 < n:
            if p1 == m:
                sorted_list.append(nums2[p2])
                p2 += 1
            elif p2 == n:
                sorted_list.append(nums1[p1])
                p1 += 1
            elif nums1[p1] < nums2[p2]:
                sorted_list.append(nums1[p1])
                p1 += 1
            else:
                sorted_list.append(nums2[p2])
                p2 += 1

        nums1[:] = sorted_list


class Solution2(object):
    def merge(self, nums1, m, nums2, n):
        nums1[m:] = nums2
        nums1.sort()

File: 01-10/Merge Sorted Array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        for (int i =0; i != n; i++) {
            nums1[m+i] = nums2[i];
        }

        Arrays.sort(nums1);
    }
}


class Solution2 {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int p1 = 0;
        int p2 = 0;
        int [] sortedList = new int[m+n];
        int cur;
        while(p1 < m || p2 < n){
            if (p1 ==m) {
                cur = nums2[p2++];
            } else if (p2 ==n) {
                cur = nums1[p1++];
            } else if (nums1[p1] < nums2[p2]) {
                cur = nums1[p1++];
            } else {
                cur = nums2[p2++];
            }
            sortedList[p1 + p2 - 1] = cur;
        }

        for (int i = 0; i < m + n; ++i) {
            nums1[i] = sortedList[i];
        }
    }
}